### DISCUSSION TITLE
Compilation error; Missing Atomic file.

### DISCUSSION THREAD

**Discussion by alexvanb28** [pschatzmann/arduino-audio-tools#1452] [Q&A] [answered]
Hey,

Im a total noobie and trying to learn but im stuck at the point where i want to test some code but i keep getting an error that im missing an Atomic library. Could someone point me to the right direction so i could start testing this awesome code!

**Comment by pschatzmann** [comment]
Not sure what you are referring to: there is no Atomic library. 
What exactly are you doing ? with what Microcontroller ?

**Reply by alexvanb28** [reply]
Im trying to play an mp3 file thru the vs1053 with an arduino and seperate micro sd card adapter. Its for school but im really struggling so far.

**Reply by alexvanb28** [reply]
![Screenshot 2024-03-17 142035](https://github.com/pschatzmann/arduino-audio-tools/assets/157494762/8b7a7dea-8e8e-4208-9c1b-ae010085ac9c)

**Comment by pschatzmann** [comment]
What Arduino Model ? I was testing [this sketch](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-vs1053/player-sd-vs1053/player-sd-vs1053.ino)  only on a UNO R4, but that was quite some time ago...

**Comment by alexvanb28** [comment]
Ohh I just have the UNO without wifi.

**Comment by alexvanb28** [comment]
Thnx for the fast replys :)

**Comment by pschatzmann** [comment, marked as answer]
A regular UNO has not enough memory for the AudioTools. 
I noticed that even the R4 does not fit into the Flash memory any more.
If you want to use this library I recommend an ESP32 or a RP2040...
